src/main/java/com/amazon/ion/impl/IonReaderNonContinuableSystem.java

```diff
@@ -46,8 +47,15 @@ final class IonReaderNonContinuableSystem implements IonReader {
      */
     IonReaderNonContinuableSystem(IonReaderContinuableCore reader) {
         this.reader = reader;
-        reader.registerIvmNotificationConsumer((x, y) -> {
-            pendingIvmSids.add(SystemSymbols.ION_1_0_SID); // TODO generalize for Ion 1.1
+        reader.registerIvmNotificationConsumer((major, minor) -> {
+            if (major != 1 || minor > 1) {
+                throw new IllegalStateException("The parser should have already thrown upon encountering this illegal IVM.");
+            }
+            if (minor == 0) {
+                pendingIvmSids.add(SystemSymbols.ION_1_0_SID);
+            } else {
+                // TODO how should this be handled for Ion 1.1?
+            }
```

Comment:
> how should this be handled for Ion 1.1?

I have some ideas, but I haven't checked them for feasibility w.r.t. backwards compatibility with existing _public_ APIs.

We don't actually need the SID of the IVM. We just need some value that represents which version it is. I don't think we should re-use `IonEncodingVersion` for this, but we could create an internal-only `VersionMarker` enum or something like that.

Alternately, maybe we could say that the system reader doesn't give you IVMs as symbols—it requires registering an IVM listener just like the regular reader.

Reply:
I think we're on the same page here. I would like if we could avoid requiring the Ion 1.1 writer to recognize SIDs as IVMs, while still allowing IVMs to be propagated from the system reader to a user-level writer. We may have to add some extra logic / callback registration between the writer and system reader to make that work, but it should be doable.